dev/src/write-batch.ts

```diff
@@ -575,12 +575,33 @@ export class WriteBatch implements firestore.WriteBatch {
       api.BatchWriteResponse
     >('batchWrite', request, tag);
 
-    return (response.writeResults || []).map((result, i) => {
+    const writeResults = response.writeResults || [];
+    let latestTimestamp = Timestamp.fromMillis(0);
+    writeResults.forEach(result => {
+      if (
+        result.updateTime &&
+        Timestamp.fromProto(result.updateTime) > latestTimestamp
+      ) {
+        latestTimestamp = Timestamp.fromProto(result.updateTime);
+      }
+    });
+
+    return writeResults.map((result, i) => {
       const status = response.status[i];
       const error = new GoogleError(status.message || undefined);
       error.code = status.code as Status;
+
+      // Since delete operations currently do not have write times, use the
+      // latest update time for deletes, or Timestamp(0,0) if the entire batch
+      // consists of deletes.
+      const isSuccessfulDelete =
+        result.updateTime === null && error.code === Status.OK;
       return new BatchWriteResult(
-        result.updateTime ? Timestamp.fromProto(result.updateTime) : null,
+        isSuccessfulDelete
```

Comment:
I don't know what is going on here :) Can you rewrite this to make it obvious? At the very least, this needs parentheses.

Reply:
I copied this style from your Transaction PR b/c I thought it seemed really cool! And then I went back and saw you changed it :(